### PR TITLE
Ensure user backup does not include ZKPs, which are very large

### PR DESCRIPTION
**Pull request by RiverRuby**
The user has options to either backup to Apple/Google wallet or copy/paste their entire store. Previously we just dumped the entire localStorage with people's sigmojis into the wallet. But now that we are also caching computed (spartan-ecdsa) proofs in localStorage, there will be too much data to add to Apple/Google wallet directly. Thus I made some more utility functions to ensure we strip the ZKPs out of localStorage before backing up.

I also added a bunch more comments for the localStorage related functions, since their uses are sort of blending together and seem to benefit from further documentation.

### CODE REVIEW

File: components/wallet/GoogleWalletButton.tsx [hunk 1]
```diff
@@ -4,24 +4,33 @@ import * as jose from "jose";
...
 export const GoogleWalletButton = () => {
-  const [serial, setSerial] = useState<string | undefined>(undefined);
-  const [firstTime, setFirstTime] = useState<boolean>(false);
   const [tempSerial, setTempSerial] = useState<string | undefined>(undefined);
   const [saveUrl, setSaveUrl] = useState<string | null>(null);
-  const [number, setNumber] = useState<number | undefined>(undefined);
   const [updated, setUpdated] = useState<boolean>(false);
 
+  const [serial, setSerial] = useState<string | undefined>(undefined);
+  const [firstTime, setFirstTime] = useState<boolean>(false);
```
Comment (AndrewCLu) [inline components/wallet/GoogleWalletButton.tsx]: Don't see a need for this. Can just use !serial

File: components/wallet/GoogleWalletButton.tsx [hunk 2]
```diff
@@ -4,24 +4,33 @@ import * as jose from "jose";
...
   loadSigmojis,
   saveBackupState,
 } from "@/lib/localStorage";
+import { Sigmoji } from "@/lib/types";
 import { PrimaryFontH4 } from "../core";
 
 export const GoogleWalletButton = () => {
-  const [serial, setSerial] = useState<string | undefined>(undefined);
-  const [firstTime, setFirstTime] = useState<boolean>(false);
   const [tempSerial, setTempSerial] = useState<string | undefined>(undefined);
```
Comment (AndrewCLu) [inline components/wallet/GoogleWalletButton.tsx]: Also no need for this, the useEffect hook to generate a save url can just run if there's no serial number already. Plus can just write `useState<string>()`

File: components/wallet/GoogleWalletButton.tsx
Comment (AndrewCLu) [inline components/wallet/GoogleWalletButton.tsx:1]: For [line 45](https://github.com/project-vinyl/nfctap.xyz/blob/459f781c40ba3b8d2c449aea73df284d4efc51cd/components/wallet/GoogleWalletButton.tsx#L45), more readable if the check is `if (serial) {return;}`

Comment (AndrewCLu) [inline components/wallet/GoogleWalletButton.tsx:1]: I assume you want to use `sigmojiWalletBackup` [here](https://github.com/project-vinyl/nfctap.xyz/blob/459f781c40ba3b8d2c449aea73df284d4efc51cd/components/wallet/GoogleWalletButton.tsx#L133)

Comment (AndrewCLu) [inline components/wallet/GoogleWalletButton.tsx:1]: Doesn't need to be done now, but [google pass url generation](https://github.com/project-vinyl/nfctap.xyz/blob/459f781c40ba3b8d2c449aea73df284d4efc51cd/components/wallet/GoogleWalletButton.tsx#L47) can be refactored into utils

**Review by AndrewCLu — COMMENTED** [review]
Mostly just state simplification/refactor comments. Only big thing is I assume you want to use the zkp-stripped backup when saving for a first time user.

File: components/wallet/GoogleWalletButton.tsx (hunk 1 above)
Comment (RiverRuby) [reply components/wallet/GoogleWalletButton.tsx]: yep! factored out

File: components/wallet/GoogleWalletButton.tsx
Comment (RiverRuby) [reply components/wallet/GoogleWalletButton.tsx:1]: Yep, let's save this for the next iteration for now since it's only in one place now

**Pull request merged**